web/pgadmin/tools/erd/static/js/erd_tool/components/MainToolBar.jsx

```diff
@@ -134,17 +133,12 @@ export function MainToolBar({preferences, eventBus, fillColor, textColor, notati
       [ERD_EVENTS.ANY_ITEM_SELECTED, (selected)=>{
         setDisableButton('drop-table', !selected);
       }],
-      [ERD_EVENTS.DIRTY, (isDirty)=>{
+      [ERD_EVENTS.DIRTY, (isDirty, data)=>{
+        const save_app_state = enableSaveToolData('ERD');
```

Comment:
only if isDirty true. Lets not trigger tool save data  on save click.

Reply:
Done.